presto-main/src/main/java/com/facebook/presto/execution/TaskExecutor.java

```diff
@@ -193,10 +213,41 @@ public synchronized String toString()
                 .toString();
     }
 
-    private synchronized void addRunnerThread()
+    private synchronized void adjustRunnerThreads(int target)
+    {
+        targetRunnerThreads = target;
+
+        int moreThreads = targetRunnerThreads - runnerThreads;
+        if (moreThreads > 0) {
+            addRunnerThreads(moreThreads);
+            runnerThreads = targetRunnerThreads;
+        }
+    }
+
+    private synchronized boolean tryStopRunner()
+    {
+        if (runnerThreads > targetRunnerThreads) {
+            runnerThreads--;
```

Comment:
@losipiuk `runnerThreads` is used before this pull request and is exported to JMX server, should we still use the same name for this variable?